File: split-bill-api/src/main/java/com/shashankbhat/splitbill/repository/BillRepository.java
package com.shashankbhat.splitbill.repository;

import com.shashankbhat.splitbill.entity.BillEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface BillRepository extends JpaRepository<BillEntity, Integer>{
    List<BillEntity> findAllByGroupId(Integer groupId);
}

File: app/src/main/java/com/shashankbhat/splitbill/repository/local/BillRepository.kt
package com.shashankbhat.splitbill.repository.local

import com.shashankbhat.splitbill.room_db.dao.BillDao
import com.shashankbhat.splitbill.room_db.entity.Bill

class BillRepository(private val billDao: BillDao) {

    suspend fun insert(bill: Bill): Long {
        return billDao.insert(bill)
    }

    suspend fun getAllBill(groupId: Int): List<Bill> {
        return billDao.getAllBill(groupId)
    }

}